src/app/blocks/grid-block/grid-block.component.ts

```diff
@@ -55,9 +57,16 @@ export class GridBlockComponent implements OnInit, OnChanges {
   }
 
   updateOutputDisplay() {
-    const defaultCols = isArray(this.model) && this.model.length > 0
-      ? Object.keys(this.model[0]).map(key => ({ headerName: key, field: key }))
-      : [];
+    let defaultCols = [];
+    if (isArray(this.model) && this.model.length > 0) {      
+      defaultCols = Object.keys(this.model[0]).map((key) => {
+          let column = { headerName: key, field: key };
+          if (this.firstRowHeaders) column.headerName = this.model[0][key];
+          return column;
+        }        
+      )
+      if (this.firstRowHeaders) this.model.shift(); // remove first row
```

Comment:
Is it more safe for this.model be explicitly set to a new value?
I wonder if mutating the model directly is the right thing to do, vs explicitly setting the model because we already have issues with how 'reactive' it is? :thinking:

Reply:
I'll switch to a clone, and not alter the underlying model here. This current implementation will cause the removal of the first row to be passed on through the chain.